### ISSUE TITLE
New Feature: "Not" block filter

### ISSUE THREAD

**Issue by laiello58** [capitalone/cloud-custodian#895]
In addition to the `and` and `or` block filters, I think it would be a good idea to have a `not` block filter, instead of having to parametrize every individual filter for that.

For example:

```
  - name: test
    description: |
      test
    resource: s3
    filters:
      - not:
        - type: some-filter
```

**Comment by kapilt** [comment]
sounds good to me, that would simplify things a bit, we have quite a few filters with boolean logic to determine match or inverse match semantics.

**Comment by scotwk** [comment]
@laiello58 - released in 0.8.23.1

https://github.com/capitalone/cloud-custodian/releases/tag/0.8.23.1
